tests/test_scalars.py

```diff
@@ -708,6 +805,293 @@ def test_timestamp() -> None:
+        ),
+        pd.Series,
+    )
+
+    check(assert_type(ts <= ts, bool), bool)
+    check(assert_type(ts <= np.datetime64(1, "ns"), bool), bool)
+    check(assert_type(ts <= dt.datetime(year=2000, month=1, day=1), bool), bool)
+    check(
+        assert_type(ts <= pd.DatetimeIndex(["2000-1-1"]), np_ndarray_bool), np.ndarray
+    )
+    check(assert_type(ts <= np_dt64_arr, np_ndarray_bool), np.ndarray)
+    check(
+        assert_type(
+            ts <= pd.Series([1, 2, 3], dtype="datetime64[ns]"), "pd.Series[bool]"
+        ),
+        pd.Series,
+    )
+
+    check(assert_type(dt.datetime(year=2000, month=1, day=1) <= ts, bool), bool)
+    check(
+        assert_type(pd.DatetimeIndex(["2000-1-1"]) <= ts, np_ndarray_bool), np.ndarray
+    )
+    check(assert_type(np_dt64_arr <= ts, np_ndarray_bool), np.ndarray)
+    check(
+        assert_type(
+            pd.Series([1, 2, 3], dtype="datetime64[ns]") <= ts, "pd.Series[bool]"
+        ),
+        pd.Series,
+    )
+
+    check(assert_type(ts == ts, bool), bool)
+    check(assert_type(ts == np.datetime64(1, "ns"), bool), bool)
+    check(assert_type(ts == dt.datetime(year=2000, month=1, day=1), bool), bool)
+    check(
+        assert_type(ts == pd.DatetimeIndex(["2000-1-1"]), np_ndarray_bool), np.ndarray
+    )
+    check(
+        assert_type(ts == np_dt64_arr, np_ndarray_bool),
+        np.ndarray,
+    )
+    check(
+        assert_type(
+            ts == pd.Series([1, 2, 3], dtype="datetime64[ns]"), "pd.Series[bool]"
+        ),
+        pd.Series,
+    )
+
+    check(assert_type(dt.datetime(year=2000, month=1, day=1) == ts, bool), bool)
+    check(
+        assert_type(pd.DatetimeIndex(["2000-1-1"]) == ts, np_ndarray_bool), np.ndarray
+    )
+
+    check(
+        assert_type(
+            pd.Series([1, 2, 3], dtype="datetime64[ns]") == ts, "pd.Series[bool]"
+        ),
+        pd.Series,
+    )
+
+    check(assert_type(ts != ts, bool), bool)
+    check(assert_type(ts != np.datetime64(1, "ns"), bool), bool)
+    check(assert_type(ts != dt.datetime(year=2000, month=1, day=1), bool), bool)
+    check(
+        assert_type(ts != pd.DatetimeIndex(["2000-1-1"]), np_ndarray_bool), np.ndarray
+    )
+    check(assert_type(ts != np_dt64_arr, np_ndarray_bool), np.ndarray)
+    check(
+        assert_type(
+            ts != pd.Series([1, 2, 3], dtype="datetime64[ns]"), "pd.Series[bool]"
+        ),
+        pd.Series,
+    )
+
+    check(assert_type(dt.datetime(year=2000, month=1, day=1) != ts, bool), bool)
+    check(
+        assert_type(pd.DatetimeIndex(["2000-1-1"]) != ts, np_ndarray_bool), np.ndarray
+    )
+    check(
+        assert_type(
+            pd.Series([1, 2, 3], dtype="datetime64[ns]") != ts, "pd.Series[bool]"
+        ),
+        pd.Series,
+    )
+
+    # Failures due to NumPy ops returning Any
+    check(
+        assert_type(  # type: ignore[assert-type]
+            np_td64_arr + ts, npt.NDArray[np.datetime64]  # type: ignore[operator]
```

Comment:
Unless we are testing for failures, we shouldn't use `# type: ignore`.  If it returns `Any`, then let's expect `Any` as the return type, and document that's what happening (as you did elsewhere)

The `ignore[operator]` one is disturbing, because if `numpy` doesn't except the addition, then we should have an `__radd__()` declaration that does accept it.

Reply:
Added an radd but there is a disagreement between Mypy and Pyright. [in #389 ]